sources/packages/backend/apps/api/src/route-controllers/assessment/_tests_/e2e/assessment.institutions.controller.getRequestedAssessmentSummary.e2e-spec.ts

```diff
@@ -106,6 +110,77 @@ describe("AssessmentInstitutionsController(e2e)-getRequestedAssessmentSummary",
       ]);
   });
 
+  it(
+    "Should not get the 'offering change' assessment requests details in the request summary " +
+      "for an eligible application when an eligible public institution user tries to access it and " +
+      "student offering change is pending with the ministry.",
+    async () => {
+      // Arrange
+
+      // Student has an application to the institution.
+      const student = await saveFakeStudent(db.dataSource);
+
+      const currentMSFAA = createFakeMSFAANumber(
+        { student },
+        {
+          state: MSFAAStates.Signed,
+        },
+      );
+      await db.msfaaNumber.save(currentMSFAA);
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          institution: collegeF,
+          institutionLocation: collegeFLocation,
+          student,
+          msfaaNumber: currentMSFAA,
+        },
+        { applicationStatus: ApplicationStatus.Completed },
+      );
+
+      // Create pending offering change.
+      const offeringRequestOfferings = createFakeOfferingRequestChange({
+        currentOffering: application.currentAssessment.offering,
+      });
+      const [, requestedOffering] = await db.educationProgramOffering.save(
+        offeringRequestOfferings,
+      );
+      const endpoint = `/institutions/assessment/student/${student.id}/application/${application.id}/requests`;
+      const institutionUserToken = await getInstitutionToken(
+        InstitutionTokenTypes.CollegeFUser,
+      );
+
+      // Act/Assert
+      await request(app.getHttpServer())
+        .get(endpoint)
+        .auth(institutionUserToken, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.OK)
+        .expect([]);
+
+      // Checking the same scenario with the ministry endpoint with a ministry user, to make sure that a fake offering change was created.
+      // Arrange.
+      const ministryEndpoint = `/aest/assessment/application/${application.id}/requests`;
```

Comment:
Testing a ministry endpoint at `assessment.institutions.controller.*spec` is against the pattern that we have been following in e2e.

I am not able to follow the reason for doing it.

Reply:
As @andrewsignori-aot mentioned https://github.com/bcgov/SIMS/pull/1976#discussion_r1213444345 and the comment stated in the code, it's not a aest endpoint check, it is just to validate the scenario, whether the offering change was actually created and that's is the reason, the institution endpoint is returning empty list (result without an offering change)

Reply:
I am not able to follow @ann-aot . To validate a scenario we are are calling an endpoint here which I feel is an E2E test for an endpoint even though the intention is to validate a scenario.